packages/contracts-bedrock/test/LivenessModule.t.sol

```diff
@@ -0,0 +1,512 @@
+        livenessModule.removeOwners(prevOwners, ownersToRemove);
+    }
+
+    /// @dev Tests if remove owners reverts if the current Safe.guard does note match the expected
+    ///      livenessGuard address.
+    function test_removeOwners_guardChanged_reverts() external {
+        address[] memory ownersToRemove = new address[](1);
+        ownersToRemove[0] = safeInstance.owners[0];
+        address[] memory prevOwners = safeInstance.getPrevOwners(ownersToRemove);
+
+        // Change the guard
+        livenessGuard = new LivenessGuard(safeInstance.safe);
+        safeInstance.setGuard(address(livenessGuard));
+
+        _warpPastLivenessInterval();
+        vm.expectRevert("LivenessModule: guard has been changed");
+        livenessModule.removeOwners(prevOwners, ownersToRemove);
+    }
+
+    function test_removeOwners_invalidThreshold_reverts() external {
+        address[] memory ownersToRemove = new address[](0);
+        address[] memory prevOwners = new address[](0);
+        uint256 wrongThreshold = safeInstance.safe.getThreshold() + 1;
+
+        vm.mockCall(
+            address(safeInstance.safe), abi.encodeCall(OwnerManager.getThreshold, ()), abi.encode(wrongThreshold)
+        );
+
+        _warpPastLivenessInterval();
+        vm.expectRevert("LivenessModule: Safe must have a threshold of 75% of the number of owners");
+        livenessModule.removeOwners(prevOwners, ownersToRemove);
+    }
+}
+
+contract LivenessModule_RemoveOwners_Test is LivenessModule_TestInit {
+    using SafeTestLib for SafeInstance;
+
+    /// @dev Tests if removing one owner works correctly
+    function test_removeOwners_oneOwner_succeeds() external {
+        uint256 ownersBefore = safeInstance.owners.length;
+        address ownerToRemove = safeInstance.owners[0];
+
+        _warpPastLivenessInterval();
+        _removeAnOwner(ownerToRemove, safeInstance.owners);
+
+        assertFalse(safeInstance.safe.isOwner(ownerToRemove));
+        assertEq(safeInstance.safe.getOwners().length, ownersBefore - 1);
+    }
+
+    /// @dev Tests if removing all owners works correctly
+    function test_removeOwners_allOwners_succeeds() external {
+        uint256 numOwners = safeInstance.owners.length;
+
+        address[] memory ownersToRemove = new address[](numOwners);
+        for (uint256 i = 0; i < numOwners; i++) {
+            ownersToRemove[i] = safeInstance.owners[i];
+        }
+        address[] memory prevOwners = safeInstance.getPrevOwners(ownersToRemove);
+
+        _warpPastLivenessInterval();
+        livenessModule.removeOwners(prevOwners, ownersToRemove);
+        assertEq(safeInstance.safe.getOwners().length, 1);
+        assertEq(safeInstance.safe.getOwners()[0], fallbackOwner);
+        assertEq(safeInstance.safe.getThreshold(), 1);
+    }
+}
+
+/// @dev A copy of LivenessModule.get75PercentThreshold as a free function to use below.
+function get75PercentThreshold(uint256 _numOwners) pure returns (uint256 threshold_) {
```

Comment:
Not super ideal to copy/paste source code like this

Reply:
I don't think it's that bad for a single line function.